utils/checks.py

```diff
@@ -0,0 +1,251 @@
+            return predicate(ctx)
+        else:
+            # If ctx is not passed, act as a decorator for command
+            return commands.check(predicate)
+
+    @classmethod
+    def role_check(
+        cls, ctx: commands.Context | disnake.ApplicationCommandInteraction, roles: list[int]
+    ) -> bool:
+        """Check if user has role that enables him to use specific command.
+
+        Returns
+        -------
+        bool
+            True if user has required role, False otherwise.
+        """
+
+        guild = ctx.bot.get_guild(config.guild_id)
+        user = guild.get_member(ctx.author.id)
+        for role in user.roles:
+            if role.id in roles:
+                return True
+        return False
+
+    @classmethod
+    def is_in_modroom(
+        cls,
+        ctx: commands.Context | disnake.ApplicationCommandInteraction = None,
+        raise_exception: bool = True,
+    ) -> bool:
+        """Check if the command is invoked in modroom"""
+
+        def predicate(ctx: commands.Context | disnake.ApplicationCommandInteraction) -> bool:
+            if ctx.channel.id == config.mod_room:
+                return True
+
+            if raise_exception:
+                raise InvalidRoomError(Messages.mod_room_only(room=config.mod_room))
+            return False
+
+        if ctx:
+            # If ctx is passed, return the result of the predicate
+            return predicate(ctx)
+        else:
+            # If ctx is not passed, act as a decorator for command
+            return commands.check(predicate)
+
+    @classmethod
+    def is_in_voteroom(
+        cls,
+        ctx: commands.Context | disnake.ApplicationCommandInteraction = None,
+        raise_exception: bool = True,
+    ) -> bool:
+        """Check if the command is invoked in voteroom"""
+
+        def predicate(ctx: commands.Context | disnake.ApplicationCommandInteraction) -> bool:
+            if ctx.channel.id == config.vote_room:
+                return True
+
+            if raise_exception:
+                raise InvalidRoomError(Messages.vote_room_only(room=config.vote_room))
+            return False
+
+        if ctx:
+            # If ctx is passed, return the result of the predicate
+            return predicate(ctx)
+        else:
+            # If ctx is not passed, act as a decorator for command
+            return commands.check(predicate)
+
+    @classmethod
+    def is_botroom(cls, inter: disnake.ApplicationCommandInteraction) -> bool:
+        """
+        Check if the interaction is in allowed channel. This is used to change response type to ephemeral.
+
+        Returns
+        -------
+        bool
+            False if user is not in DMs | thread | allowed channel. Otherwise True.
+        """
+
+        # DMs with bot
+        if inter.guild is None:
+            return False
+
+        # allow threads in channels
+        if isinstance(inter.channel, disnake.Thread):
```

Comment:
Wouldn't it be better to only allow threads in allowed channels?
```suggestion
        if isinstance(inter.channel, disnake.Thread):
            if inter.channel.parent_id in config.allowed_channels():
                return False
```